From: Utkarsh Dixit <utkarsh.extc@gmail.com>
Date: Wed, 12 Jan 2022 14:14:21 +0530
Subject: [PATCH] Update agent.go (#1)

* Update agent.go

* Update jaeger.go

* Update agent.go

* Update agent.go

diff --git a/agent.go b/agent.go
index 7ae2b31..d08c9dc 100644
--- a/agent.go
+++ b/agent.go
@@ -18,6 +18,7 @@ import (
 	"fmt"
 	"io"
 	"net"
+	"context"
 
 	"github.com/uber/jaeger-client-go/thrift"
 	"github.com/uber/jaeger-client-go/thrift-gen/agent"
@@ -73,8 +74,8 @@ func newAgentClientUDP(hostPort string, maxPacketSize int) (*agentClientUDP, err
 // EmitBatch implements EmitBatch() of Agent interface
 func (a *agentClientUDP) EmitBatch(batch *jaeger.Batch) error {
 	a.thriftBuffer.Reset()
-	a.client.SeqId = 0 // we have no need for distinct SeqIds for our one-way UDP messages
-	if err := a.client.EmitBatch(batch); err != nil {
+// 	a.client.SeqId = 0 // we have no need for distinct SeqIds for our one-way UDP messages
+	if err := a.client.EmitBatch(context.Background(), batch); err != nil {
 		return err
 	}
 	if a.thriftBuffer.Len() > a.maxPacketSize {
diff --git a/jaeger.go b/jaeger.go
index 2fd480b..735a154 100644
--- a/jaeger.go
+++ b/jaeger.go
@@ -24,6 +24,7 @@ import (
 	"io/ioutil"
 	"log"
 	"net/http"
+	"context"
 
 	"github.com/uber/jaeger-client-go/thrift"
 	"github.com/uber/jaeger-client-go/thrift-gen/jaeger"
@@ -358,7 +359,7 @@ func (e *Exporter) uploadCollector(batch *jaeger.Batch) error {
 
 func serialize(obj thrift.TStruct) (*bytes.Buffer, error) {
 	buf := thrift.NewTMemoryBuffer()
-	if err := obj.Write(thrift.NewTBinaryProtocolTransport(buf)); err != nil {
+	if err := obj.Write(context.Background(),thrift.NewTBinaryProtocolTransport(buf)); err != nil {
 		return nil, err
 	}
 	return buf.Buffer, nil
